### ISSUE TITLE
`Source` option not working as expected

### ISSUE THREAD

**Issue by simondel** [PowerShellOrg/cChoco#44]
I'm having some issues with the `Source` option in the `cChocoPackageInstaller`. We have a private Chocolatey repository which we use for some packages. I'm working a package right now which has the same name as a public package, my config is like this:

``` powershell
cChocoSource Privaterepo
{
    Name = 'privaterepo'
    Source = 'https://our.private.repo.net/nuget'
}

cChocoPackageInstaller HipChat
{
    Name = 'hipchat'
    Source = 'privaterepo'
    DependsOn = "[cChocoInstaller]InstallChocolatey", "[cChocoSource]Privaterepo"
}
```

During testing our private hipchat package had a lower version (0.0.6) than the public package (2.2.1164) and when applying the DCS configuration, the public package was installed instead of our private package.

I would expect that our private package would be installed here since the command

```
choco install hipchat --source=privaterepo
```

**does** install our private version.

**Comment by ferventcoder** [comment]
When specifying a source, Chocolatey should only look at that source.

@simondel I think the provider here may not be passing through quite correctly. 

I'm guessing the following may work as a workaround for now:

``` powershell
cChocoPackageInstaller HipChat
{
    Name = 'hipchat'
    Source = 'https://our.private.repo.net/nuget'
    DependsOn = "[cChocoInstaller]InstallChocolatey", "[cChocoSource]Privaterepo"
}
```

**Comment by ferventcoder** [comment]
If not, it's likely a bug in this provider.

**Comment by simondel** [comment]
Thanks for the quick response! I initially used your suggestion and then moved to a cChocoSource resource when that didn't work (and because I was specifying the url more than once in my DSC config).
So I think this is a bug, unfortunately.

**Comment by ferventcoder** [comment]
This looks like the culprit:

https://github.com/PowerShellOrg/cChoco/blob/b44cd3bfa0c5b423f344847015b37b903d6ab165/DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1#L79-L92

**Comment by DamianBis** [comment]
Yup. the issue is that instead of running the install passing through the -source command to chocolatey it is just adding the source to chocolatey.

**Comment by javydekoning** [comment]
Indeed, the code mentioned above is incorrect. If #50 get's merged I'll continue work on this, seems like a rather easy thing to fix.

**Comment by travis-sobeck** [comment]
Is there a timeline for getting this resolved, its still broken?
Thanks

**Comment by ferventcoder** [comment]
This is still not yet resolved, however I'm going to set it up for being fixed in 2.4.0.

**Comment by gmatusko** [comment]
Let us know when you have a fix for this, we're experiencing the same issue when trying to install packages from various sources, internal and external.  Working around it by adding the sources to the config using cChocoSource but would be nice if this worked as expected

**Comment by travis-sobeck** [comment]
As an aside, I found that as long as my package was not named exactly the same as a public one, it works fine.

**Comment by ferventcoder** [comment]
This looks to be fixed with #84.

**Comment by ferventcoder** [comment]
This is now fixed and will go into the next release of cChoco
